library/alloc/src/slice.rs

```diff
@@ -1016,13 +1021,15 @@ where
     // Finally, `hole` gets dropped. If the shorter run was not fully consumed, whatever remains of
     // it will now be copied into the hole in `v`.
 
-    unsafe fn get_and_increment<T>(ptr: &mut *mut T) -> *mut T {
+    #[inline(always)]
+    fn get_and_increment<T>(ptr: &mut *mut T) -> *mut T {
         let old = *ptr;
         *ptr = unsafe { ptr.offset(1) };
         old
     }
 
-    unsafe fn decrement_and_get<T>(ptr: &mut *mut T) -> *mut T {
+    #[inline(always)]
+    fn decrement_and_get<T>(ptr: &mut *mut T) -> *mut T {
```

Comment:
I believe all these were unsafe for a reason; without that you can call them with arguments that can be unsound without any warning that you've done so.

Reply:
I was getting this warning so I removed it:
![image](https://user-images.githubusercontent.com/6633152/140375991-aecfb9e3-a5fe-4039-bc48-c4cffa130c9c.png)

Reply:
I believe the purpose of unsafe functions is to warn the potential user of the api provided by the function.  Since these functions are not meant as APIs but are utility functions local to the sort, I had decided to remove the `unsafe` **on** the function rather that remove the `unsafe` **in** the function.  Leaving it in the function makes it cleared exactly what is unsafe.  Am I wrong to do that?

Reply:
it depends, it may be more important to indicate the unsafety where these helpers are used rather than what parts of the helpers are unsafe.

Reply:
I've put the `unsafe`s back.